Azure Pipleines Environement Agent Info uses FilePath instead of DirectoryPath

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

5.0.0

### Operating system

macOS

### Operating system architecture

Arm64

### CI Server

Azure DevOps

### What are you seeing?

When referring to the properties `AzurePipelinesAgentInfo` they are returning `FilePath` instead of `DirectoryPath` for the properties:

- BuildDirectory
- HomeDirectory
- ToolsDirectory
- WorkingDirectory

### What is expected?

I would expect the properties to be `DirectoryPath` so you can easily use `Combine` and other methods to create a path. You have to convert the `FilePath` to `DirectoryPath` manually first.

This was fixed on Build info previously. See: https://github.com/cake-build/cake/issues/3590

### Steps to Reproduce

Calling something like: `AzurePipelines.Environment.Agent.HomeDirectory.Combine(".sonarqube/out/");` will give you:

```
error CS1061: 'FilePath' does not contain a definition for 'Combine' and no accessible extension method 'Combine' accepting a first argument of type 'FilePath' could be found (are you missing a using directive or an assembly reference?)
```

### Output log

_No response_

--- a/src/Cake.Common/Build/AzurePipelines/Data/AzurePipelinesAgentInfo.cs
+++ b/src/Cake.Common/Build/AzurePipelines/Data/AzurePipelinesAgentInfo.cs
@@ -28,7 +28,7 @@ namespace Cake.Common.Build.AzurePipelines.Data
         /// The local path on the agent where all folders for a given build definition are created.
         /// </value>
         /// <example><c>c:\agent\_work\1</c>.</example>
-        public FilePath BuildDirectory => GetEnvironmentString("AGENT_BUILDDIRECTORY");
+        public DirectoryPath BuildDirectory => GetEnvironmentString("AGENT_BUILDDIRECTORY");
 
         /// <summary>
         /// Gets the directory the agent is installed into. This contains the agent software.
@@ -38,7 +38,7 @@ namespace Cake.Common.Build.AzurePipelines.Data
         /// The directory the agent is installed into.
         /// </value>
         /// <example><c>c:\agent\</c>.</example>
-        public FilePath HomeDirectory => GetEnvironmentString("AGENT_HOMEDIRECTORY");
+        public DirectoryPath HomeDirectory => GetEnvironmentString("AGENT_HOMEDIRECTORY");
 
         /// <summary>
         /// Gets the working directory for this agent.
@@ -46,7 +46,7 @@ namespace Cake.Common.Build.AzurePipelines.Data
         /// <value>
         /// The working directory for this agent.
         /// </value>
-        public FilePath WorkingDirectory => GetEnvironmentString("AGENT_WORKFOLDER");
+        public DirectoryPath WorkingDirectory => GetEnvironmentString("AGENT_WORKFOLDER");
 
         /// <summary>
         /// Gets the ID of the agent.
@@ -98,7 +98,7 @@ namespace Cake.Common.Build.AzurePipelines.Data
         /// <value>
         /// The task directory.
         /// </value>
-        public FilePath ToolsDirectory => GetEnvironmentString("AGENT_TOOLSDIRECTORY");
+        public DirectoryPath ToolsDirectory => GetEnvironmentString("AGENT_TOOLSDIRECTORY");
 
         /// <summary>
         /// Gets a value indicating whether the current agent is a Microsoft hosted agent.

